src/router/Route.ts

```diff
@@ -29,6 +33,19 @@ export default class Route extends Component<IRouteProps, any> {
 				onEnter({ props: this.props, router });
 			});
 		}
+
+		const { getComponent } = this.props;
+		if (getComponent) {
+			Promise.resolve().then(() => {
+				getComponent({ props: this.props, router }, this.onComponentResolved);
+			});
+		}
+	}
+
+	onComponentResolved = (error, component) => {
```

Comment:
Can we make this

`private _onComponentResolved(...)`

Also, why have the unused error variable?